src/methods/mapzonal.jl

```diff
@@ -0,0 +1,130 @@
+"""
+    mapzonal(reducer, operator, data::RasterStack; of, kw...)
+
+Think of this like `zonal`, but a named tuple gets sent to the `operator`, 
+and `reducer` gets the result of `operator.(data[mask])`.
+
+# Arguments
+
+- `reducer`: any function that can consume an iterable as its only argument, such as `sum` or `Statistics.mean`.  Can also be `identity` or any other user-defined function!
+- `operator`: any function that can consume a named tuple as its only argument, such as `mean` or `std`.  Applied on the set of values from RasterStacks, what you would get when you index into a RasterStack like `rasterstack[1, 1]`.
+- `data`: a `RasterStack`
+- `of`: a `DimTuple`, `Extent`, $OBJ_ARGUMENT
+
+# Keywords
+$GEOMETRYCOLUMN_KEYWORD
+These can be used when `of` is or contains (a) GeoInterface.jl compatible object(s):
+
+- `shape`: Force `data` to be treated as `:polygon`, `:line` or `:point`, where possible.
+- `boundary`: for polygons, include pixels where the `:center` is inside the polygon,
+    where the line `:touches` the pixel, or that are completely `:inside` inside the polygon.
+    The default is `:center`.
+- `progress`: show a progress bar, `true` by default, `false` to hide..
+- `skipmissing`: wether to apply `f` to the result of `skipmissing(A)` or not. If `true`
+    `f` will be passed an iterator over the values, which loses all spatial information.
+    if `false` `f` will be passed a masked `Raster` or `RasterStack`, and will be responsible
+    for handling missing values itself. The default value is `true`.
+
+# Example
+
+"""
+function mapzonal(reducer, operator, data::RasterStackOrArray; of, kw...)
+    _mapzonal(reducer, operator, data, of; kw...)
+end
+
+function _mapzonal(reducer, operator, x, ext::Extents.Extent; skipmissing = true)
+    cropped = crop(x; to = ext, touches = true)
+    prod(size(cropped)) > 0 || return missing
+    # We can't use skipmissing here, since it doesn't work on rasterstacks
```

Comment:
We should just make this work, but it's just not totally clear what the rule is - any missing or all missing

Reply:
In here I've implemented the "all missing" variant but it's up for debate

Reply:
I guess the question is, does `skipmissing` mean "skip all values that are missing in general" or "skip all values that are missing because of my mask"?  The first is the most useful because there can be functions that don't tolerate missings